### PR TITLE
Fixed incorrect variable name in `LibGit2.upstream`

### LINKED ISSUE #16337: UndefVarError in libgit2 error
``` julia
=> Package RunTests cannot be updated.
GitError(Code:ERROR, Class:Net, Unsupported URL protocol)

=> Package MbedTLS cannot be updated.
UndefVarError: repo_ptr_ptr not defined
 in upstream(::Base.LibGit2.GitReference) at ./libgit2/reference.jl:141
```

I assume this should be `ref_ptr_ptr`?

### PR DESCRIPTION
**Pull request by wildart**
Fixed typos in `LibGit2.lookup_branch` and `LibGit2.upstream`

### CODE REVIEW

**Comment by tkelman** [comment]
This should have caused a test failure. Why didn't it?

**Comment by wildart** [comment]
Because, it is happens when error is returned. We never tested error cases.

**Comment by tkelman** [comment]
Can we figure out a test that would have hit this?

**Comment by yuyichao** [comment]
Please have a more descriptive commit message and PR title....

Git short log and git blame also only show the first line of the commit message so please put the description in the first line too....

File: test/libgit2.jl [hunk 1]
```diff
@@ -248,15 +248,20 @@ mktempdir() do dir
...
+                    @test LibGit2.isbranch(brref)
+                    @test LibGit2.name(brref) == "refs/heads/master"
+                    @test LibGit2.shortname(brref) == "master"
+                    @test LibGit2.ishead(brref)
+                    @test LibGit2.upstream(brref) == nothing
+                    @test repo.ptr == LibGit2.owner(brref).ptr
+                    @test brnch == "master"
+                    @test LibGit2.headname(repo) == "master"
+                    LibGit2.branch!(repo, test_branch, string(commit_oid1), set_head=false)
+                    @test LibGit2.lookup_branch(repo, test_branch, true) == nothing
```
Comment (tkelman) [inline test/libgit2.jl]: and the other branches? what about `upstream`?

**Comment by tkelman** [comment]
Looks good, though some or all of theae functions could use docstrings explaining what they're for.

**Comment by wildart** [comment]
I'll do documentation for internals of LibGit2 module in a separate PR.

**Pull request merged**